pkg/version/version.go

```diff
@@ -3,18 +3,32 @@ package version
 import (
 	"fmt"
 	"runtime"
+	"runtime/debug"
 )
 
-// Vcs is is the commit hash for the binary build.
-var Vcs string
-
-// Timestamp is the date for the binary build.
-var Timestamp string
-
 // Version is the gatekeeper version.
 var Version string
 
-// GetUserAgent returns a user agent of the format: gatekeeper/<version> (<goos>/<goarch>) <vcs>/<timestamp>.
+// GetUserAgent returns a user agent of the format: gatekeeper/<version> (<goos>/<goarch>) <vcsrevision><-vcsdirty>/<vcstimestamp>.
 func GetUserAgent() string {
-	return fmt.Sprintf("gatekeeper/%s (%s/%s) %s/%s", Version, runtime.GOOS, runtime.GOARCH, Vcs, Timestamp)
+	vcsrevision := "unknown"
+	vcstimestamp := "unknown"
+	vcsdirty := ""
+
+	if info, ok := debug.ReadBuildInfo(); ok {
```

Comment:
I like this PR!

QQ, what happens when `!ok` to the version here? even this would even happen at all.

Reply:
looks like it might be `!ok` if it's built without mod support or parsing fails https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/runtime/debug/mod.go;l=20;drc=178080740c1bc33f2c7f164504eedc24210bbf1e;bpv=1;bpt=1

Reply:
from the code it looks like it defaults to "unknown/unknown"

Reply:
should the build fail instead of returning unknown/unknown?

Reply:
this wouldn't happen in our CI

Reply:
Since this is mostly for our CI, I'm ok leave it as unknown for now unless someone raises an issue.

Reply:
IMO unknown/unknown is fine in order to avoid unnecessary brittleness for people hacking on G8r. If we wanted to ensure our builds aren't affected, we could test that the flag returns an expected result?